channel/app.go

```diff
@@ -34,11 +35,13 @@ type (
 		// returns false.
 		Def() wallet.Address
 
-		// DecodeData decodes data specific to this application. This has to be
-		// defined on an application-level because every app can have completely
-		// different data; during decoding the application needs to be known to
-		// know how to decode the data.
-		DecodeData(io.Reader) (Data, error)
+		// NewData returns an instance of data specific to this application.
+		// This has to be defined on an application-level because every app can
+		// have completely different data. This instance should be used for
+		// unmarshalling the binary representation of the data.
+		//
+		// It is zero-value is safe to use.
```

Comment:
```suggestion
		// It is zero-value and is safe to use.
```
Is that what you wanted to say?

Reply:
I have rephrased it in 27f6382f . Does it look fine now ?